spec/features/reservations_spec.rb

```diff
@@ -0,0 +1,330 @@
+          Time.zone.today + (@eq_model.maximum_checkout_length + 1).days
+        update_cart_due_date(bad_due_date)
+        # request catalog since our update cart methods only return the cart
+        # partial from the JS update (so the 'Reserve' link wouldn't be there)
+        visit root_path
+        click_link 'Reserve', href: new_reservation_path
+
+        # set reserver to current user if unset, check confirmation page
+        reserver = reserver ? reserver : @current_user
+        expect(page).to have_content reserver.name
+        expect(page).to have_content Time.zone.today.to_s(:long)
+        expect(page).to have_content bad_due_date.to_s(:long)
+        expect(page).to have_content @eq_model.name
+        expect(page).to have_content 'File Reservation Request'
+        find(:xpath, "//textarea[@id='reservation_notes']").set 'Because'
+        click_button 'Submit Request'
+
+        # check that reservation was created with correct dates
+        # can't use for_eq_model since it calls finalized
+        query = Reservation.where(equipment_model_id: @eq_model.id)
+                .for_reserver(reserver.id)
+        expect(query.count).to eq(1)
+        expect(query.first.start_date).to eq(Time.zone.today)
+        expect(query.first.due_date).to eq(bad_due_date)
+        expect(query.first.approval_status).to eq('requested')
+      end
+    end
+
+    shared_examples 'can create failing reservation' do |reserver|
+      let(:reserver) { reserver }
+      str = reserver ? 'for other user successfully' : 'successfully'
+
+      it str do
+        visit root_path
+        change_reserver(reserver) if reserver
+        add_item_to_cart(@eq_model)
+        update_cart_start_date(Time.zone.today)
+        # violate length validation
+        bad_due_date =
+          Time.zone.today + (@eq_model.maximum_checkout_length + 1).days
+        update_cart_due_date(bad_due_date)
+        # request catalog since our update cart methods only return the cart
+        # partial from the JS update (so the 'Reserve' link wouldn't be there)
+        visit root_path
+        click_link 'Reserve', href: new_reservation_path
+
+        # set reserver to current user if unset, check confirmation page
+        reserver = reserver ? reserver : @current_user
+        expect(page).to have_content reserver.name
+        expect(page).to have_content Time.zone.today.to_s(:long)
+        expect(page).to have_content bad_due_date.to_s(:long)
+        expect(page).to have_content @eq_model.name
+        expect(page).to have_content 'Create Reservation'
+        expect(page).to have_content 'Please be aware of the following errors:'
+        find(:xpath, "//textarea[@id='reservation_notes']").set 'Because'
+        click_button 'Finalize Reservation'
+
+        # check that reservation was created with correct dates
+        # can't use for_eq_model since it calls finalized
+        query = Reservation.where(equipment_model_id: @eq_model.id)
```

Comment:
It would only work for the valid reservations, but not for the requests. See my [earlier comment](https://github.com/YaleSTC/reservations/pull/1219/files#r28663431), I'm fine to explicitly call `finalized` where we're currently just calling `for_eq_model` and then make the `for_eq_model` scope *only* a function of the equipment model.